internal/driver/driver.go

```diff
@@ -140,17 +162,21 @@ func (d *Driver) Start() error {
 	// Make sure the paths of existing devices are up-to-date.
 	go d.RefreshAllDevicePaths()
 
-	d.wg.Add(1)
-	go d.StartRTSPCredentialServer()
+	if d.enableRtspServer {
+
+		d.wg.Add(1)
+		go d.StartRTSPCredentialServer()
```

Comment:
I know this isn't related to your code, but if you wouldn't mind tweaking this a bit. I prefer the wg.Add and wg.Done to be in close sight of each other. Right now, the wg.Done is in the `StartRTSPCredentialServer` func. I would remove it from there, and do this:

```go
		d.wg.Add(1)
		go func() {
				defer wg.Done()
				d.StartRTSPCredentialServer()
		}
```

this helps to make sure every added waitgroup item is accounted for (and in other situations prevent from having to pass the wg).